charts/DataTab.tsx

```diff
@@ -20,49 +21,59 @@ export class DataTab extends React.Component<{
     // Here's where the actual CSV is made
     @computed get csvBlob() {
         const { chart } = this.props
-        const { vardata } = chart
+        const yearIsDayVar = chart.yearIsDayVar
+        const dayIndexedCSV = yearIsDayVar ? true : false
 
         const dimensions = chart.data.filledDimensions.filter(
             d => d.property !== "color"
         )
-        const entitiesUniq = sortBy(
-            uniq(flatten(dimensions.map(d => d.entitiesUniq)))
-        ) as string[]
-        const yearsUniq = sortBy(
-            uniq(flatten(dimensions.map(d => d.yearsUniq)))
-        ) as number[]
+        const uniqueEntitiesAcrossDimensions =
+            chart.uniqueEntitiesAcrossDimensions
+
+        // only get days if chart has a day-indexed variable, else get years across dimensions
+        const indexingYears = sortBy(
+            dayIndexedCSV
```

Comment:
Have you thought about splitting this into some smaller methods? A general csvBlob() method, that does the check is dayIndexedCSV, and then 2 methods one that generates the CSV for the Year style charts and the other for the day style charts? Might then be easier to unit test those 2. Or might not be necessary.

Reply:
So I thought hard about this one as I agree the method has at least three different points where it checks if the current chart is day-based or year-based, suggesting there might be two different branches of logic that could be separated into two different methods. I decided against it for now, for the following reasons:
1. The amount of shared logic exceeds the amount of unshared logic for the two kinds of charts. `csvBlob()` has a particular logic for iterating through entities and years to generate the final data download, and I think separating the two kinds of charts might require me to reproduce that shared logic across two different functions. The checks for day-based vs year-based charts are small modifications on formatting, not on how the CSV is basically constructed. I'm not sure I can improve readability here significantly without created reproduced logic.
2. I could create a `csvConfig` object with `titleRow` and other parameters that differ between day-based and year-based charts, but I'm not sure I could simplify `csvBlob()` that much except for getting rid of the lines populating `titleRow` and `indexingYears`.

If you have the time and desire, and would be interested in suggesting an alternative, I'm very open to learning!